generic/tests/ping.py

```diff
@@ -1,133 +1,182 @@
 import logging
+import aexpect
 
 from autotest.client.shared import error
 from autotest.client import utils
 
 from virttest import utils_test
 from virttest import utils_net
+from virttest import error_context
 
 
-@error.context_aware
+@error_context.context_aware
 def run(test, params, env):
     """
     Ping the guest with different size of packets.
 
     1) Login to guest
-    2) Ping test on nic(s) from host
+    2) Ping test on nic(s) from host - default_ping/multi_nics
         2.1) Ping with packet size from 0 to 65507
-        2.2) Flood ping test
-        2.3) Ping test after flood ping, Check if the network is still alive
-    3) Ping test from guest side, packet size is from 0 to 65507
-       (win guest is up to 65500) (Optional)
+        2.2) Ping with exceed packet size (65508)
+        2.3) Flood ping test
+        2.4) Ping test after flood ping, Check if the network is still alive
+    3) Ping test from guest side to external host - ext_host
+        3.1) Ping with packet size from 0 to 65507 (win guest is up to 65500)
+        3.2) Ping with exceed packet size (65508)
+        3.3) Flood ping test
+        3.4) Ping test after flood ping, Check if the network is still alive
 
     :param test: QEMU test object.
     :param params: Dictionary with the test parameters.
     :param env: Dictionary with test environment.
     """
-    def _get_loss_ratio(output):
+    def _check_result(status, output):
         if params.get("strict_check", "no") == "yes":
             ratio = utils_test.get_loss_ratio(output)
             if ratio != 0:
-                raise error.TestFail("Loss ratio is %s" % ratio)
+                test.fail("Loss ratio is %s" % ratio)
+        if status != 0:
+            test.fail("Ping failed, status: %s,"
+                      " output: %s" % (status, output))
 
     timeout = int(params.get("login_timeout", 360))
     ping_ext_host = params.get("ping_ext_host", "no") == "yes"
 
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
-    error.context("Login to guest", logging.info)
+    error_context.context("Login to guest", logging.info)
     session = vm.wait_for_login(timeout=timeout)
 
     # most of linux distribution don't add IP configuration for extra nics,
     # so get IP for extra nics via pre_cmd;
     if params.get("pre_cmd"):
         session.cmd(params["pre_cmd"], timeout=600)
 
+    ext_host = "www.redhat.com"
```

Comment:
I would suggest we can directly ping `www.redhat.com`.
Now it firstly assigns `www.redhat.com` to`ext_host`, then it is trying  to get another IP to ping.
If it can not get the IP, it will ping `www.redhat.com` instead, which is redundant.

Reply:
It will find the nearest route to ping, if cannot find, will ping www.redhat.com.  If ping www.redhat.com directly, maybe need many routes to get www.redhat.com, will influence the network performance and lead to timeout. And there are also some security problems.